src/manifest.c

```diff
@@ -428,13 +430,25 @@ int match_manifests(struct manifest *m1, struct manifest *m2)
 			file3->is_config = file1->is_config;
 			file3->is_state = file1->is_state;
 			file3->is_boot = file1->is_boot;
+			/* ghost deleted boot files */
+			file3->is_ghosted = file1->is_boot;
 
-			if (!file1->is_deleted) {
+			if (file3->is_ghosted) {
+				/* if the new file is ghosted, preserve the hash and rename
+				 * information and set the last_change version to the current
+				 * version. */
 				file3->last_change = m2->version;
-			} else {
+				file3->is_rename = file1->is_rename;
+				hash_assign(file1->hash, file3->hash);
+			} else if (file1->is_deleted) {
```

Comment:
Only `file3->last_change` is different between these two conditions; consider condensing them into one block, with a smaller conditional (or trinary) for `file3->last_change` only.

Reply:
Or we even use `m2->version` for everything else... I think this can be condensed a lot, into single-line conditionals.